stablehlo/tests/stablehlo_legalize_deprecated_ops.mlir

```diff
@@ -0,0 +1,76 @@
+// RUN: stablehlo-opt --stablehlo-legalize-deprecated-ops --split-input-file --verify-diagnostics %s | FileCheck %s
+
+// CHECK-LABEL: op_dot
+func.func @op_dot(%arg0: tensor<2x3xf32>,
+                 %arg1: tensor<3x?xf32>) -> tensor<2x?xf32> {
+  // CHECK: stablehlo.dot_general %arg0, %arg1, contracting_dims = [1] x [0] : (tensor<2x3xf32>, tensor<3x?xf32>) -> tensor<2x?xf32>
+  %0 = "stablehlo.dot"(%arg0, %arg1) : (tensor<2x3xf32>, tensor<3x?xf32>) -> tensor<2x?xf32>
```

Comment:
Maybe add a few cases with vectors?

Reply:
(By vectors, specifically I mean 1-dimensional tensors, as opposed to the 2-dimensional tensors used here.)